### PR TITLE
Add instance-pool API

### PR DESCRIPTION
**Pull request by DaanRademaker**
I would like to add the instance-pool api to the custom resource so this can also be used from CDK.

### CODE REVIEW

File: deploy-lambda/src/databricks_cdk/job.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class AutoScale(BaseModel):
 class AwsAttributes(BaseModel):
     first_on_demand: int = 1
     availability: str = "SPOT_WITH_FALLBACK"
-    zone_id: str
+    zone_id: Optional[str] = None
```
Comment (DaanRademaker) [inline deploy-lambda/src/databricks_cdk/job.py:28]: I added this as optional because when deploying a job with an instance pool this needs to be None.

Comment (ffinfo) [reply deploy-lambda/src/databricks_cdk/job.py:28]: 👍

**Comment by DaanRademaker** [comment]
@ffinfo , I added some simple "happy flow" tests for instance pools.

File: typescript/src/resources/instance-pools/instance-pools.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import {CustomResource} from "aws-cdk-lib";
+import {Construct} from "constructs";
+
+export interface InstancePoolAwsAttributes {
+    first_on_demand?: number
```
Comment (ffinfo) [inline typescript/src/resources/instance-pools/instance-pools.ts:5]: Does not exists

**Review by ffinfo — APPROVED** [review]

**Pull request merged**